proto/catalog.proto

```diff
@@ -156,6 +161,9 @@ message SinkFormatDesc {
   optional plan_common.EncodeType key_encode = 4;
   // Secret used for format encode options.
   map<string, secret.SecretRef> secret_refs = 5;
+
+  // ref connection for schema registry
+  optional uint32 connection_id = 6;
```

Comment:
I think this should be added to `SinkParam` & `SinkDesc` instead, side-by-side with the Sink's `properties`.

Reply:
> We do solve connection ref when creating sink, so the props in Sink.properties contains the ones from connection catalog.

I see. So here we expand all the connections into concrete parameters **on creation**, right? This is okay to me, but in this way, I think we should not have a `connection_id` here in `SinkFormatDesc` as well, because all `connection`-related things are eliminated on creation.

(It seems `SinkFormatDesc::connection_id` was indeed not used anywhere?)

Reply:
Similar to the source side, the connector part and schema registry part will have independent connection, just like individual secret_refs. 

we are offering syntax like

```
create source s(...) with (
  connector = 'kafka',
  connection = <connector connection>, ...
) format ... encode ... (
  connection = <schema registry connection / glue connection>
)
```

Reply:
Yeah, I know. What I mean is: should it be resolved and eliminated before converting into a proto message? 

> We do solve connection ref when creating sink, so the props in `Sink.properties` contains the ones from connection catalog.

Reply:
> IIRC, secret ref doesn't keep the resolved plaintext secret at all. It always resolves whenever using a secret.

Oh, here is a some gap on `resolve`. 
You can ref to `resolve_connection_ref_and_secret_ref`, resolve here means extracting `xxx = secret <secret name>` from "with props" to `PbSecretRef (secret_id)`. We will do the same op for secrets coming from both connection catalog and with props.